src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+        public void SendCommand(ICommand command)
+        {
+            if (command == null)
+            {
+                throw new ArgumentNullException(nameof(command));
+            }
+
+            if (command.GetBytes() == null || command.GetBytes().Length == 0)
+            {
+                throw new ArgumentException("The command did not contain any bytes to send.");
+            }
+
+            SendCommand(command.GetBytes());
+
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        public void SendCommand(Ssd1351Command command)
+        {
+            SendSPI(new byte[] { (byte)command }, true);
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        /// <param name="data">Byte array to send as parameters for the command.</param>
+        public void SendCommand(Ssd1351Command command, byte[] data)
+        {
+
+            SendCommand(command);
+
+            if (data != null && data.Length > 0)
+            {
+                SendSPI(data);
+
+                // detect certain commands that may alter the state of the device. This is done as the 
+                // SPI device cannot read registers from the ssd1351 and so changes need to be captured
+                switch (command)
+                {
+                    // capture changes to the colour depth and colour sequence
+                    case Ssd1351Command.SetRemap:
+                        _colorSequence = (ColorSequence)((data[0] >> 2) & 0x01);
+                        _colorDepth = (ColorDepth)((data[0] >> 6) & 0x03);
+                        break;
+                }
+
+            }
+
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="data">Byte array to send as a combination of command in the first byte
+        /// and following with parameters for the command.</param>
+        public void SendCommand(byte[] data)
+        {
+            Span<byte> buffer = data;
+
+            if(data == null || data.Length == 0)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (data.Length == 0)
+            {
+                throw new ArgumentException("Data array is empty", nameof(data));
+            }
+
+            SendCommand((Ssd1351Command)data[0], buffer.Slice(1).ToArray());
+        }
+
+        /// <summary>
+        /// Send data to the display controller.
+        /// </summary>
+        /// <param name="data">The data to send to the display controller.</param>
+        public void SendData(byte[] data)
+        {
+            const int StackThreshold = 512;
```

Comment:
Do you mind explaining why did we pick this constant as the stack threshold?

Reply:
@joperezr 

I can't remember right now but it I cant think of why this should be needed after reading one of the MS blogs so I have removed the conditional stackalloc and now just use it in all cases in the latest PR updated